package/MDAnalysis/analysis/density.py

```diff
@@ -394,8 +395,18 @@ def _prepare(self):
                 logger.warning(msg)
             # Generate a copy of smin/smax from coords to later check if the
             # defined box might be too small for the selection
-            smin = np.min(coord, axis=0)
-            smax = np.max(coord, axis=0)
+            try:
+                smin = np.min(coord, axis=0)
+                smax = np.max(coord, axis=0)
+            except ValueError as err:
+                msg = ("No atoms in AtomGroup at input time frame. "
+                       "Grid for density could not be automatically"
+                       " generated. A user defined grid was found"
+                       " and is being used as the density grid")
+                warnings.warn(msg)
+                logger.warning(msg)
+                smin = self._gridcenter
+                smax = self._gridcenter
             # Overwrite smin/smax with user defined values
             smin, smax = self._set_user_grid(self._gridcenter, self._xdim,
```

Comment:
@IAlibay what are your thoughts on emitting a warning if `self._gridcenter` is not set but `self._xdim, sel._ydim` etc is not? I know its mentioned in docs, but IMO it looks easy to have your grid being ignored as incomplete even if not the intention.

Reply:
I'm not sure I understand what you mean here. If X/y/zdim are not set you'll get a ValueError in _set_user_grid?

Reply:
I meant the other way round. I could be missing something, but AFAIK if `gridcenter` is not set but `xdim`, `ydim` and `zdim` are set we never reach `_set_user_grid()` and `xdim`, `ydim` and `zdim` are ignored silently? 

Envisioning a warning like `"user defined grid coordinates are set but no grid centre is present, 
 user defined grid coordinates are being ignored."`?

Reply:
I see, maybe we can change the `if` check to `any` that covers gridcenter and the dims? Doing so should trigger the type checking in _set_user_grid.

Ideally I'd prefer not adding extra error/warning emitting unless we move all this to _set_user_grid and revamp the error checking there. If we go down that route, I'd prefer we deal with this in a separate PR though.

If we choose to go down the `any` route, which should be the simplest to deal with, then I'm happy with us doing this here, but we'll need an extra test probably.

Reply:
Idea: if none of them are set, default to box size if box exists?

Reply:
That's dangerous imho, not only because it adds an extra default condition, but also because box size isn't an easy / cheap thing to define under NPT conditions. It'd essentially require an extra loop across the trajectory to get the max box dimensions etc... We can certainly have that option but imho it'd be better to have it as a non-default.

Re xyzdim I'd vote that our current approach is the simplest, fail unless you provide numbers. 

I agree that not setting grid center when xyzdim  are set shouldnt be a silent default either.

Reply:
@hmacdope do you want to raise a separate issue for this? I say we go ahead with merging this PR as is and we can do the necessary changes to further validate user inputs for the grids?

Reply:
Ack I'll raise an issue.